product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/vfs/VFSClient.java

```diff
@@ -189,6 +189,80 @@ public void ftpFileDownload(String sourceFilePath, File targetFilePath) throws I
         return Arrays.asList(fileDirectoryList);
     }
 
+    /**
+     * Function to create new directories in an FTP server
+     *
+     * @param dirPath directory path
+     * @throws IOException if changing directory or creating new directory fails
+     */
+    public void makeDirectories(String dirPath) throws IOException {
+        dirPath = dirPath.startsWith("/") ? dirPath.substring(1) : dirPath;
+        String[] pathElements = dirPath.split("/");
+        if (pathElements.length > 0) {
+            for (String singleDir : pathElements) {
+                boolean dirExisted = ftpClient.changeWorkingDirectory(singleDir);
+                if (!dirExisted) {
+                    boolean dirCreated = ftpClient.makeDirectory(singleDir);
+                    if (dirCreated) {
+                        log.info("Created directory : " + singleDir);
+                        ftpClient.changeWorkingDirectory(singleDir);
+                    } else {
+                        log.info("Could not directory : " + singleDir);
```

Comment:
This should be error log and exit the loop since rest of the attempts will also fail

Reply:
Done